includes/abstracts/abstract-wc-order.php

```diff
@@ -780,14 +780,38 @@ public function get_item_count( $item_type = '' ) {
 	}
 
 	/**
-	 * Get an order item object, based on it's type.
+	 * Get an order item object, based on it's ID. The item must belong to the current
+	 * order. If the item cannot be found or it doens't belong to current order
+	 * FALSE will be returned.
 	 *
 	 * @since  3.0.0
 	 * @param  int $item_id
 	 * @return WC_Order_Item
 	 */
 	public function get_item( $item_id ) {
-		return WC_Order_Factory::get_order_item( $item_id );
+		foreach ( $this->items as $items ) {
+			if ( ! empty( $items[ $item_id ] ) ) {
+				return $items[ $item_id ];
+			}
+		}
+
+		if ( ! $this->get_id() ) {
+			return false;
+		}
+
+		// The item was not found in memory, we load in memory all the items related to this order
```

Comment:
> Why do we need to load ALL item types here rather than doing what get_order_item in the factory does and get the type of the item being loaded?
 
Great feedback, I improved it in ea9c573.